keps/sig-scheduling/1819-scheduler-extender/README.md

```diff
@@ -0,0 +1,299 @@
+    - [Filter](#filter)
+    - [Prioritize](#prioritize)
+    - [Bind](#bind)
+    - [Preempt](#preempt)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubernetes scheduler schedules based on resources managed by Kubernetes.
+Scheduling based on opaque resource counting helps extending this further. But
+when there is a need for contextual scheduling for resources managed outside of
+kubernetes, there is no mechanism to do it today.
+
+The proposal is to make kubernetes scheduler extensible by adding the capability
+to make http calls out to another endpoint to help achieve this functionality.
+
+## Motivation
+
+There are three ways to add new scheduling rules to Kubernetes:
+
+- Update existing or add new scheduler plugins and recompiling
+- Implementing your own scheduler process that runs instead of, or alongside of, the standard Kubernetes scheduler
+- Implementing a "scheduler extender" process that the standard Kubernetes scheduler calls out to as a final pass when making scheduling decisions
+
```

Comment:
+1, perhaps rather than frame it as pros/cons, we can present it as what each approach is more suitable for.
Scheduler extender: if you don't care about latency and cache consistency and don't want to maintain your own build of the default scheduler, then extender is likely the better choice. Plugins: if you care about performance and want maximum ability to customize the scheduler's behavior, then developing a new plugin is likely the better choice.

Reply:
added a paragraph for this

Reply:
> +1, perhaps rather than frame it as pros/cons, we can present it as what each approach is more suitable for.
> Scheduler extender: if you don't care about latency and cache consistency and don't want to maintain your own build of the default scheduler, then extender is likely the better choice. Plugins: if you care about performance and want maximum ability to customize the scheduler's behavior, then developing a new plugin is likely the better choice.

I just saw your comments, SG to me. I will make an update soon

Reply:
I've updated the last paragraph to reflect this. The old sentence "This approach is needed for use cases where scheduling decisions need to be made on resources not directly managed by the standard Kubernetes scheduler. " is also out of date because this approach is not the only way to make scheduling decisions on external resources.